### PR TITLE
Implement proper analog button deadzone

### PR DESCRIPTION
**Pull request by rsn8887**
## Related Issues

This implements a proper analog deadzone for analog buttons while keeping the radial scaled deadzone for analog sticks.

This fixes analog buttons when non-zero deadzone is selected. See also
https://github.com/libretro/RetroArch/commit/e0b59f13e20d8b293463635f0cb01e12b01ea38e#diff-7797ae0e7813aded66f109c460a562da

## Reviewers

@dobyrch 
@twinaphex

### CODE REVIEW

**Comment by rsn8887** [comment]
@dobyrch : This should be a better way to fix your issue with analog buttons, can you test maybe? I don't have analog buttons :)

**Comment by rsn8887** [comment]
Needs testing currently.

**Comment by dobyrch** [comment]
Unfortunately this doesn't work for me.  I'm getting the same behavior as before, buttons mapped to the right analog Y-axis only trigger when the deadzone is set to 0.

**Comment by rsn8887** [comment]
@dobyrch I force-pushed a change that might fix it. Do you mind testing? Thank you!!!!!

**Comment by dobyrch** [comment]
That did the trick! The latest commit (3c3ac7b7beaa) looks good to me.

**Comment by rsn8887** [comment]
@twinaphex: Ready to merge!

**Comment by rsn8887** [comment]
The earlier method was still not quite correct, because it assumed that x/y axes are always numbered a certain way, which can be different depending on input driver, joypad mapping etc. Can you test one last (hopefully last) time, @dobyrch ? THANKS!!!!

**Comment by dobyrch** [comment]
With the latest change, if the deadzone is set to any value other than 0, only right and down (X+ and Y+) work.

**Comment by rsn8887** [comment]
Another try ;) thanks for all your help, @dobyrch

**Comment by dobyrch** [comment]
No problem, thank you for putting in the work to get this fixed!
745fb4b40ca9 works for me.

**Comment by rsn8887** [comment]
Sweet!

@twinaphex, tested and ready to merge whenever you have time! Thanks!

File: retroarch.c [hunk 1]
```diff
@@ -16067,10 +16032,13 @@ int16_t input_joypad_analog(const input_device_driver_t *drv,
...
 
          /* Analog button. */
-         /* no deadzone/sensitivity correction for analog buttons currently */
          if (drv->axis)
-            res = abs(drv->axis(joypad_info.joy_idx, axis));
-
+         {
+            float normal_mag = 0.0f;
+            if (configuration_settings->floats.input_analog_deadzone)
+               normal_mag = abs((1.0f / 0x7fff) * drv->axis(joypad_info.joy_idx, axis));
```
Comment (ToadKing) [inline retroarch.c]: This was already merged, but should this be `fabs` instead? Getting warnings in compiles from it.

Comment (rsn8887) [reply retroarch.c]: Yes this was already spotted and fixed here https://github.com/libretro/RetroArch/commit/79ce99715697aafdc88a422039eb246a27484ce4

**Pull request merged**